lib/akismet.js

```diff
@@ -73,17 +75,15 @@ var Akismet = function(options) {
       referrer             : options.referrer || options.referer || ''
     })
     .set('User-Agent', this.userAgent)
-    .end(function(err, res) {
-      if (err) cb(err, null);
-      else if (res.text == 'true') cb(null, true);
-      else if (res.text == 'false') cb(null, false);
-      else if (res.header['x-akismet-debug-help']) 
-        cb(new Error(res.header['x-akismet-debug-help']), null);
-      else if (res.text == 'invalid')
-        cb(new Error('Invalid API key'), null);
-      else 
-        cb(new Error(res.text), null);
-    });
+    .endAsync()
+    .then(function(res) {
+      if (res.text == 'true') return true;
```

Comment:
@andapopovici `res` should always be defined, unless there is a request error in which case `then` will not be called at all. `res.text` will always be defined for requests that respond with text or JSON data. 

The reason there is no `err` parameter with this promise implementation is because `then()` is for handling success conditions only. If there is an error, the promise will be rejected and `then()` not called. People who use promises will be able to call `catch()` on the returned promise to handle error conditions, and people who use callbacks will see any error as the `err` parameter in their callback.